Correcting peak VALU Roofline profiling and analysis by removing FP8 VALU and BF16 VALU benchmarking.

ggottipa-amd opened the pull request:
## Motivation

FP8 and BF16 data types need to be removed from Peak VALU Roofline profiling and analysis.

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->

Correction of Peak VALU Benchmarking in roofline.csv and roofline plots.

## Technical Details

FP8 and BF16 removed from PEAK_OPS_DATATYPES.

## Test Plan

Verified the roofline plots on running the following

../src/rocprof-compute profile -n occupancy_t1 --device 0 --roof-only --roofline-data-type FP8 FP32 BF16 -- ./occupancy  
../src/rocprof-compute analyze -p workloads/occupancy_t1/MI300X_A1/ --roofline-data-type FP8 BF16 --gui
../src/rocprof-compute analyze -p workloads/occupancy_t1/MI300X_A1/ --roofline-data-type FP8 FP32 --gui


## Test Result

FP8 and BF16 - Peak VALU **NOT** seen.
FP32 - Peak VALU seen.

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

ggottipa-amd committed:
Removing FP8 from peak VALU datatypes - PEAK_OPS_DATATYPES.

vedithal-amd reviewed (approved):
Please make sure roofline.csv does not have FP8 FLOPs

vedithal-amd commented:
How does the roofline benchmark calculate and dump FP8 FLOPs in roofline.csv if FP8 data type only uses MFMA units? This seems like a bug in the roofline benchmark and should be fixed as part of this PR to ensure FP8 FLOPs is not present in roofline.csv

cfallows-amd commented:
> How does the roofline benchmark calculate and dump FP8 FLOPs in roofline.csv if FP8 data type only uses MFMA units? This seems like a bug in the roofline benchmark and should be fixed as part of this PR to ensure FP8 FLOPs is not present in roofline.csv

Yes this is a two-part issue, we need to adjust from rocm-amdgpu-bench as well. I will have offline (off-of-github) discussion. For additional steps.

vedithal-amd commented:
In that case we won't be able to fix benchmark utility to not show FP8 FLOPs in this PR but a separate PR in amdgpu bench repo

cfallows-amd commented:
> In that case we won't be able to fix benchmark utility to not show FP8 FLOPs in this PR but a separate PR in amdgpu bench repo

Plan is to do the change in rocm-amdgpu-bench and bring the rebuilt binaries into this PR. Bins and this change should go together as they directly correlate; safer for any reason we need to revert.

cfallows-amd committed:
Roofline binaries from rocm-amdgpu-bench generated 10/22.
https://github.com/ROCm/rocm-amdgpu-bench/commit/2113ef1f5eada8a4a6e44e6d07fd6abac9b0a3f8
Bins include change that removes FP8 and BF16 peak VALU benchmarks.
Built and tested on rhel8, azl3, ubuntu22.04, sles15sp6.

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

cfallows-amd commented:
I have added the required roofline binaries from rocm-amdgpu-bench that include the commit for removal of FP8 and BF16 VALU benchmarks. This change means both BF16 VALU and FP8 VALU are no longer being written into roofline.csv

TODO:
- please include removal of BF16 peak VALU (keep BF16 MFMA) in rocprofiler-compute, same as done for FP8, including any references in the docs if needed.
- add to changelog, best to mark this as a resolved issue, something along the lines of "Corrected peak VALU Roofline profiling and analysis by removing FP8 VALU and BF16 VALU benchmarking." Please add Pratik to the review once this is completed as well so docs team gives us the approval.

cfallows-amd reviewed (changes requested):
See comment with TODO for additional ask with BF16 removal. Title of the PR and the primary comment can be edited to reflect that we are doing the same work for an additional datatype.

ggottipa-amd committed:
Similar change for BF16.

cfallows-amd committed:
Re-committing the bins

accidentally copied over bins from the wrong folder earlier, caught by Gowthami during testing.

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

cfallows-amd commented:
Gowthami has caught issue with bins still running bench; I originally copied over bins from wrong folder. I have committed correct bins now. This PR should be good to go after a quick re-verification. Thanks

ggottipa-amd committed:
Updated changelog

ggottipa-amd committed:
gersemi fix

cfallows-amd reviewed (approved):
LGTM- good to commit to branch assuming rocprof-compute run looks proper with correct (lack of fp8 and bf16 VALU) outputs. Thanks!

prbasyal-amd commented on projects/rocprofiler-compute/CHANGELOG.md:
```diff
@@ -63,6 +63,7 @@ Full documentation for ROCm Compute Profiler is available at [https://rocm.docs.
 * Improved Roofline Benchmarking by updating the `flops_benchmark` calculation.
 
 ### Resolved issues
+* Corrected peak VALU Roofline profiling and analysis by removing FP8 VALU and BF16 VALU benchmarking.
```
```suggestion
* Corrected peak VALU Roofline profiling and analysis by removing `FP8` VALU and `BF16` VALU benchmarking.
```

prbasyal-amd replied:
This entry is listed under ROCm 7.1.1, which is already locked for a new update. Can you confirm the associated ROCm version and have it listed under "ROCm Compute Profiler <rocprof-compute version> for < ROCm-version>" (Eg. ROCm Compute Profiler 3.5.0 for ROCm 7.10.0.)

prbasyal-amd reviewed (commented):
Feedback on changelog

ggottipa-amd committed:
Changelog corrected.

ggottipa-amd committed:
Changelog fix.

JeniferC99 committed:
Merge branch 'develop' into users/ggottipa-amd/fix-peak-rooflinefp8

cfallows-amd reviewed (changes requested):
Please rebase to latest (changelog has an added 7.2.0 cutoff section now) and add this to the 7.2.0 section to be picked up in an RC build.

ggottipa-amd committed:
Adding this to the 7.2.0 section to be picked up in an RC build.

ggottipa-amd committed:
Merge branch 'develop' into users/ggottipa-amd/fix-peak-rooflinefp8

ggottipa-amd committed:
Merge branch 'develop' into users/ggottipa-amd/fix-peak-rooflinefp8

cfallows-amd committed:
Moving changelog entry into unreleasesd section - team reconfirmed cutoff date after I requested this change so I am just quickly correcting my mistake in my ask.

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

cfallows-amd reviewed (approved):
Made a mistake requesting 7.2 entry, team clarified the dates with devops team to find the cut already happened. Moved the entry to unreleased section to correct my mistake in my ask. Changelog is good to go now.
Overall good for push once passing checks. Thanks
